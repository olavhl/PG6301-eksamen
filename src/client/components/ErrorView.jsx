import React from "react";

export function ErrorView({ error, reload }) {
  return (
    <div className={"cont"}>
      <div>
        Something went wrong
        <br />
        {error.toString()}
      </div>
      {reload && <button onClick={reload}>Try again</button>}
    </div>
  );
}
