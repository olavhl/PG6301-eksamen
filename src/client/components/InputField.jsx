import React from "react";

export function InputField({ label, onChangeValue, value }) {
  return (
    <div>
      <label>
        {label}:{" "}
        <input
          type="text"
          value={value}
          onChange={(e) => onChangeValue(e.target.value)}
        />
      </label>
    </div>
  );
}
