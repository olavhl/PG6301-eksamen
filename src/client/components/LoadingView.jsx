import React from "react";

export function LoadingView() {
  return <div className={"cont"}>Loading...</div>;
}
