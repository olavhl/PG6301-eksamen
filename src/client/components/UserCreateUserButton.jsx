import { Link } from "react-router-dom";
import React from "react";

export function UserCreateUserButton() {
  return (
    <div>
      <h3>You dont have any users/friends :(</h3>
      <Link to={"/user/create"}>
        <button>Create user</button>
      </Link>
    </div>
  );
}
