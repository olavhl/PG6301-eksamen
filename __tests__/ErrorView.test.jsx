import React from "react";
import ReactDOM from "react-dom";
import { ErrorView } from "../src/client/components/ErrorView";

describe("error view", () => {
  it("show error view", () => {
    const container = document.createElement("div");
    const error = "Nope, not working";
    ReactDOM.render(<ErrorView error={error} />, container);

    expect(container.querySelector("div").textContent).toEqual(
      "Something went wrongNope, not working"
    );
    expect(container.innerHTML).toMatchSnapshot();
  });
});
