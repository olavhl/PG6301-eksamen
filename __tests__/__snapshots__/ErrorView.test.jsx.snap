// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error view show error view 1`] = `"<div>Something went wrong<br>Nope, not working</div>"`;
